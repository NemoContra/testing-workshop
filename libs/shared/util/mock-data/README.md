# shared-util-mock-data

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test shared-util-mock-data` to execute the unit tests via [Jest](https://jestjs.io).

## Running lint

Run `nx lint shared-util-mock-data` to execute the lint via [ESLint](https://eslint.org/).
