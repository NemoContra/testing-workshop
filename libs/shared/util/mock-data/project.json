{
  "$schema": "../../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/shared/util/mock-data/src",
  "projectType": "library",
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/shared/util/mock-data/**/*.ts"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/shared/util/mock-data"],
      "options": {
        "jestConfig": "libs/shared/util/mock-data/jest.config.ts",
        "passWithNoTests": true
      }
    }
  },
  "tags": ["scope:shared", "type:util"]
}
