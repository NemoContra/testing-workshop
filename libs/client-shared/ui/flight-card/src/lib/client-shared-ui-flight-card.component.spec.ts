import { createHostFactory } from '@ngneat/spectator/jest';
import { ClientSharedUiFlightCardComponent } from './client-shared-ui-flight-card.component';
import {
  createSpyObserver,
  testIdSelector,
} from '@testing-workshop/shared/util/test-helpers';
import { flights } from '@testing-workshop/shared/util/mock-data';

jest.mock('@testing-workshop/shared/util/common', () => ({
  ...jest.requireActual('@testing-workshop/shared/util/common'),
  createRandomId: jest.fn(
    (prefix) => `${prefix}-c1bfb6ed-5503-4549-8914-686553a408e1`
  ),
}));

describe('ClientSharedUiFlightCardComponent', () => {
  const createComponent = createHostFactory(ClientSharedUiFlightCardComponent);

  const selectors = {
    selectButton: testIdSelector('select-button'),
    deselectButton: testIdSelector('deselect-button'),
  } as const;

  it.each`
    flight        | selected
    ${flights[0]} | ${true}
    ${flights[0]} | ${false}
    ${flights[1]} | ${true}
    ${flights[1]} | ${false}
  `(
    'should render card with flight from $flight.from to $flight.to and selected: $selected',
    ({ flight, selected }) => {
      const spectator = createComponent(`<flight-card></flight-card>`, {
        props: {
          flight,
          selected,
        },
      });

      expect(spectator.fixture).toMatchSnapshot();
    }
  );

  it('should select and deselect a card', () => {
    const spectator = createComponent(`<flight-card></flight-card>`, {
      props: {
        flight: flights[0],
        selected: false,
      },
    });

    const spyObserver = createSpyObserver();
    spectator.output('selectedChange').subscribe(spyObserver);

    spectator.click(selectors.selectButton);
    expect(spectator.fixture).toMatchSnapshot('selected');

    spectator.click(selectors.deselectButton);
    expect(spectator.fixture).toMatchSnapshot('deselected');

    expect(spyObserver).toMatchSnapshot('selectedChange');
  });
});
