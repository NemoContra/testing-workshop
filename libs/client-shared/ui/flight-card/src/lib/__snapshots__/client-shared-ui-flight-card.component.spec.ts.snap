// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientSharedUiFlightCardComponent should render card with flight from Graz to Hamburg and selected: false 1`] = `
<ng-component
  __ngContext__={[Function Number]}
>
  <flight-card>
    <div
      class="card"
      data-is-selected="false"
      data-tid="flight-card"
      id="flight-card-c1bfb6ed-5503-4549-8914-686553a408e1"
      style="background-color: white;"
    >
      <div
        class="header"
      >
        <h2
          class="title"
          data-tid="header-text"
        >
          <span
            data-tid="flight-from"
          >
            Graz
          </span>
           - 
          <span
            data-tid="flight-to"
          >
            Hamburg
          </span>
        </h2>
      </div>
      <div
        class="content"
      >
        <p>
           Flight-No.: #
          <span
            data-tid="flight-no"
          >
            2
          </span>
        </p>
        <p>
           Date: 
          <span
            data-tid="flight-date"
          >
            October 12, 2020 at 6:22:52 AM GMT+0
          </span>
        </p>
        <p>
           Delayed: 
          <span
            data-tid="flight-delayed"
          >
            no
          </span>
        </p>
        <p>
          <button
            class="btn btn-default"
            data-tid="select-button"
          >
             Select 
          </button>
        </p>
      </div>
    </div>
  </flight-card>
</ng-component>
`;

exports[`ClientSharedUiFlightCardComponent should render card with flight from Graz to Hamburg and selected: true 1`] = `
<ng-component
  __ngContext__={[Function Number]}
>
  <flight-card>
    <div
      class="card"
      data-is-selected="true"
      data-tid="flight-card"
      id="flight-card-c1bfb6ed-5503-4549-8914-686553a408e1"
      style="background-color: rgb(204, 197, 185);"
    >
      <div
        class="header"
      >
        <h2
          class="title"
          data-tid="header-text"
        >
          <span
            data-tid="flight-from"
          >
            Graz
          </span>
           - 
          <span
            data-tid="flight-to"
          >
            Hamburg
          </span>
        </h2>
      </div>
      <div
        class="content"
      >
        <p>
           Flight-No.: #
          <span
            data-tid="flight-no"
          >
            2
          </span>
        </p>
        <p>
           Date: 
          <span
            data-tid="flight-date"
          >
            October 12, 2020 at 6:22:52 AM GMT+0
          </span>
        </p>
        <p>
           Delayed: 
          <span
            data-tid="flight-delayed"
          >
            no
          </span>
        </p>
        <p>
          <button
            class="btn btn-default"
            data-tid="deselect-button"
          >
             Remove 
          </button>
        </p>
      </div>
    </div>
  </flight-card>
</ng-component>
`;

exports[`ClientSharedUiFlightCardComponent should render card with flight from Graz to Mallorca and selected: false 1`] = `
<ng-component
  __ngContext__={[Function Number]}
>
  <flight-card>
    <div
      class="card"
      data-is-selected="false"
      data-tid="flight-card"
      id="flight-card-c1bfb6ed-5503-4549-8914-686553a408e1"
      style="background-color: white;"
    >
      <div
        class="header"
      >
        <h2
          class="title"
          data-tid="header-text"
        >
          <span
            data-tid="flight-from"
          >
            Graz
          </span>
           - 
          <span
            data-tid="flight-to"
          >
            Mallorca
          </span>
        </h2>
      </div>
      <div
        class="content"
      >
        <p>
           Flight-No.: #
          <span
            data-tid="flight-no"
          >
            1
          </span>
        </p>
        <p>
           Date: 
          <span
            data-tid="flight-date"
          >
            October 15, 2020 at 6:22:52 AM GMT+0
          </span>
        </p>
        <p>
           Delayed: 
          <span
            data-tid="flight-delayed"
          >
            no
          </span>
        </p>
        <p>
          <button
            class="btn btn-default"
            data-tid="select-button"
          >
             Select 
          </button>
        </p>
      </div>
    </div>
  </flight-card>
</ng-component>
`;

exports[`ClientSharedUiFlightCardComponent should render card with flight from Graz to Mallorca and selected: true 1`] = `
<ng-component
  __ngContext__="0"
>
  <flight-card>
    <div
      class="card"
      data-is-selected="true"
      data-tid="flight-card"
      id="flight-card-c1bfb6ed-5503-4549-8914-686553a408e1"
      style="background-color: rgb(204, 197, 185);"
    >
      <div
        class="header"
      >
        <h2
          class="title"
          data-tid="header-text"
        >
          <span
            data-tid="flight-from"
          >
            Graz
          </span>
           - 
          <span
            data-tid="flight-to"
          >
            Mallorca
          </span>
        </h2>
      </div>
      <div
        class="content"
      >
        <p>
           Flight-No.: #
          <span
            data-tid="flight-no"
          >
            1
          </span>
        </p>
        <p>
           Date: 
          <span
            data-tid="flight-date"
          >
            October 15, 2020 at 6:22:52 AM GMT+0
          </span>
        </p>
        <p>
           Delayed: 
          <span
            data-tid="flight-delayed"
          >
            no
          </span>
        </p>
        <p>
          <button
            class="btn btn-default"
            data-tid="deselect-button"
          >
             Remove 
          </button>
        </p>
      </div>
    </div>
  </flight-card>
</ng-component>
`;

exports[`ClientSharedUiFlightCardComponent should select and deselect a card: deselected 1`] = `
<ng-component
  __ngContext__={[Function Number]}
>
  <flight-card>
    <div
      class="card"
      data-is-selected="false"
      data-tid="flight-card"
      id="flight-card-c1bfb6ed-5503-4549-8914-686553a408e1"
      style="background-color: white;"
    >
      <div
        class="header"
      >
        <h2
          class="title"
          data-tid="header-text"
        >
          <span
            data-tid="flight-from"
          >
            Graz
          </span>
           - 
          <span
            data-tid="flight-to"
          >
            Mallorca
          </span>
        </h2>
      </div>
      <div
        class="content"
      >
        <p>
           Flight-No.: #
          <span
            data-tid="flight-no"
          >
            1
          </span>
        </p>
        <p>
           Date: 
          <span
            data-tid="flight-date"
          >
            October 15, 2020 at 6:22:52 AM GMT+0
          </span>
        </p>
        <p>
           Delayed: 
          <span
            data-tid="flight-delayed"
          >
            no
          </span>
        </p>
        <p>
          <button
            class="btn btn-default"
            data-tid="select-button"
          >
             Select 
          </button>
        </p>
      </div>
    </div>
  </flight-card>
</ng-component>
`;

exports[`ClientSharedUiFlightCardComponent should select and deselect a card: selected 1`] = `
<ng-component
  __ngContext__={[Function Number]}
>
  <flight-card>
    <div
      class="card"
      data-is-selected="true"
      data-tid="flight-card"
      id="flight-card-c1bfb6ed-5503-4549-8914-686553a408e1"
      style="background-color: rgb(204, 197, 185);"
    >
      <div
        class="header"
      >
        <h2
          class="title"
          data-tid="header-text"
        >
          <span
            data-tid="flight-from"
          >
            Graz
          </span>
           - 
          <span
            data-tid="flight-to"
          >
            Mallorca
          </span>
        </h2>
      </div>
      <div
        class="content"
      >
        <p>
           Flight-No.: #
          <span
            data-tid="flight-no"
          >
            1
          </span>
        </p>
        <p>
           Date: 
          <span
            data-tid="flight-date"
          >
            October 15, 2020 at 6:22:52 AM GMT+0
          </span>
        </p>
        <p>
           Delayed: 
          <span
            data-tid="flight-delayed"
          >
            no
          </span>
        </p>
        <p>
          <button
            class="btn btn-default"
            data-tid="deselect-button"
          >
             Remove 
          </button>
        </p>
      </div>
    </div>
  </flight-card>
</ng-component>
`;

exports[`ClientSharedUiFlightCardComponent should select and deselect a card: selectedChange 1`] = `
Object {
  "complete": [MockFunction complete],
  "error": [MockFunction error],
  "next": [MockFunction next] {
    "calls": Array [
      Array [
        true,
      ],
      Array [
        false,
      ],
    ],
    "results": Array [
      Object {
        "type": "return",
        "value": undefined,
      },
      Object {
        "type": "return",
        "value": undefined,
      },
    ],
  },
}
`;
